src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -182,6 +184,62 @@ public String extractCaseIdFromMetadata(Object storedDocument) {
 
     }
 
+    @Override
+    public boolean patchDocumentMetadata(CaseDocumentMetadata caseDocumentMetadata, String serviceAuthorization, String userId) {
+        try {
+            LinkedMultiValueMap<String, Object> bodyMap = new LinkedMultiValueMap<>();
+            HttpHeaders headers = new HttpHeaders();
+
+            prepareRequestForAttachingDocumentToCase(caseDocumentMetadata, serviceAuthorization, userId, bodyMap, headers);
+            HttpEntity<LinkedMultiValueMap<String, Object>> requestEntity = new HttpEntity<>(bodyMap, headers);
+
+            restTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+            restTemplate.exchange(documentURL.concat("/documents"), HttpMethod.PATCH, requestEntity, Void.class);
+
+        } catch (RestClientException ex) {
+            LOG.error("Exception while attaching a document to case : " + ex);
+            throw ex;
+        }
+        return true;
+    }
+
+    private void prepareRequestForAttachingDocumentToCase(CaseDocumentMetadata caseDocumentMetadata,
+                                                      String serviceAuthorization,
+                                                      String userId,
+                                                      LinkedMultiValueMap<String, Object> bodyMap,
+                                                      HttpHeaders headers) {
+
+        for (Document document : caseDocumentMetadata.getDocuments()) {
+            ResponseEntity responseEntity = getDocumentMetadata(UUID.fromString(document.getId()));
+            if (responseEntity.getStatusCode().equals(HttpStatus.OK) && null != responseEntity.getBody()) {
+                StoredDocumentHalResource resource = (StoredDocumentHalResource) responseEntity.getBody();
+
+                String hashcodeFromStoredDocument = ApplicationUtils.generateHashCode(document.getId()
+                                                       .concat(resource.getMetadata().get("jurisdictionId"))
+                                                       .concat(resource.getMetadata().get("caseTypeId")));
+                if (!hashcodeFromStoredDocument.equals(document.getHashToken())) {
+                    throw new ResourceNotFoundException(String.format(": Document %s does not exists in DM Store", document.getId()));
+                }
+            }
+
+            Map<String, String> metadataMap = new HashMap<>();
+            metadataMap.put("jurisdictionId", caseDocumentMetadata.getJurisdictionId());
+            metadataMap.put("caseId", caseDocumentMetadata.getCaseId());
+            metadataMap.put("caseTypeId", caseDocumentMetadata.getCaseTypeId());
```

Comment:
We don't need to put entry for the JID and caseTypeId as its already there. this call should only update caseID.